sklearn/externals/funcsigs.py

```diff
@@ -20,8 +20,8 @@
 __all__ = ['BoundArguments', 'Parameter', 'Signature', 'signature']
 
 
-_WrapperDescriptor = type(type.__call__)
```

Comment:
funcsigs is a backport of Python 3 function introspection tools for Python 2 compat. It is maintained here:

https://github.com/testing-cabal/funcsigs/

We just embed a vendored copy in `sklearn.externals`. @Daetalus could you please submit your change in a PR upstream? If it get accepted there we can merge this PR here. I would rather not have our vendored funcsigs diverge from upstream otherwise it will become a pain to maintain.